### DISCUSSION TITLE
Touch Repeat directive not working as expected on key pressed

### DISCUSSION THREAD

**Discussion by sylcha** [quasarframework/quasar#12772] [General - Components / Directives / etc]
Hi everyone!

I'm currently working on a project and I have an issue using the `v-touch-repeat` directive. 

To experiment this issue, please, go to this [page](https://heuristic-bartik-b85793.netlify.app/), click on the top left button to open the menu and choose the **Nombres** app and then enter a number at the keyboard and press ENTER.

When you click on the right or left arrow buttons, the number goes right or left respectively. Good. Now, I would like to implement the same behavior when I press the left or right arrow *keys*. I add a `left` and `right` modifier to the `v-touch-repeat` directive to the button but it does not act as expected. The arrow keys are detected only when you have pressed the matching button before. For example, if you click the left arrow *button*, the number goes left and immediately after that if you press the left arrow *key* the number goes left too (which is good) but when you press the right arrow *key* nothing happens...

Does anyone has any idea how to fix this (code of the page is given below...)?
Thanks in advance,

Sylvain

-----

# Code of the page (which is inserted in a `MainLayout` component)

```html
<template>
  <q-page padding class="bg-primary">
    <div class="column justify-center" style="min-height: inherit">
      <!-- <div class="row justify-center"> -->
      <div id="canvasPlace" class="col-8">
        <div id="p5Canvas"></div>
      </div>
      <!-- </div> -->
      <div class="row justify-center text-white">
        <div class="column" style="width: 150px">
          <div class="col self-center">
            <q-btn
              v-touch-repeat:0:1000.mouse.left="deplacerNombreGauche"
              flat
              dense
              round
              icon="ti-arrow-left"
              aria-label="Gauche"
            />
          </div>
        </div>
        <div class="column" style="width: 150px">
          <div class="col self-center">
            <q-btn
              flat
              dense
              round
              icon="ti-reload"
              aria-label="Retour"
              @click="ramenerPositionInitiale"
            />
          </div>
        </div>
        <div class="column" style="width: 150px">
          <div class="col self-center">
            <q-btn
              v-touch-repeat:0:1000.mouse.right="deplacerNombreDroite"
              flat
              dense
              round
              icon="ti-arrow-right"
              aria-label="Droite"
            />
          </div>
        </div>
      </div>
    </div>
  </q-page>
</template>

<script>
import p5 from "p5";
import { sketch } from "../js/ZoneNombre.js";
import { mapState, mapGetters, useStore } from "vuex";
let d;
export default {
  name: "NombresPage",
  setup() {
    const store = useStore();
    return {
      store,
    };
  },
  computed: {
    ...mapState("nombre", ["nombreCourant"]),
    ...mapGetters("nombre", ["nbCourant"]),
  },
  watch: {
    nombreCourant(newValue, oldValue) {
      d.nb.valeur = newValue;
    },
  },
  mounted() {
    d = new p5(sketch, "p5Canvas");
    setTimeout(() => {
      d.nb.valeur = this.store.state.nombre.nombreCourant;
    }, 500);
  },
  unmounted() {
    d.remove();
  },
  methods: {
    deplacerNombreDroite() {
      d.nb.diminuer();
    },
    deplacerNombreGauche() {
      d.nb.augmenter();
    },
    ramenerPositionInitiale() {
      // d.nb.valeur = d.nb.valeurInitiale;
    },
  },
};
</script>

```

**Comment by t-manan** [comment]
I have been trying to do a similar thing using v-touch-repeat.mouse.down in q-select tag , but it is not working for me also,

Any help will be appreciated

**Reply by pdanpdan** [reply]
we need a codepen

**Reply by t-manan** [reply]
Ah thanks @pdanpdan , I got what i am looking for.

**Comment by pdanpdan** [comment]
I don't understand - on the left button you only listen for left on on the right button only for right

https://codepen.io/pdanpdan/pen/NWMvgLj?editors=1010
